cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -329,6 +329,47 @@ protected VersionIdentifier getInstalledVersion(Path toolPath) {
     }
   }
 
+  /**
+   * @return the installed edition of this tool or {@code null} if not installed.
+   */
+  public String getInstalledEdition() {
+
+    return getInstalledEdition(this.context.getSoftwarePath().resolve(getName()));
+  }
+
+  /**
+   * @param toolPath the installation {@link Path} where to find currently installed tool. The name of the parent
+   *        directory of the real path corresponding to the passed {@link Path path} must be the name of the edition.
+   * @return the installed edition of this tool or {@code null} if not installed.
+   */
+  public String getInstalledEdition(Path toolPath) {
+
+    if (!Files.isDirectory(toolPath)) {
+      this.context.debug("Tool {} not installed in {}", getName(), toolPath);
+      return null;
+    }
+    try {
+      String edition = toolPath.toRealPath().getParent().getFileName().toString();
+      return edition;
+    } catch (IOException e) {
+      this.context.debug("When calling getInstalledEdition() for tool " + getName()
+          + " calling toRealPath().getParent() on " + toolPath + " failed. Returning null.");
```

Comment:
Isn't this a bit too detailed?

Reply:
Shouldn't we throw an exception then instead?
Otherwise I agree that we should use your rephrased message.

Reply:
Btw. we should replace the concatentation with `{}`

Reply:
Is it faster, saver, or just coding convention within our code?

Reply:
"faster" and coding convention I'd say.

Reply:
Okey. Similar to the `getInstalledVersion()` I would use a `IllegalStateException` but then I must stay with  concatenation instead of `{}`.